test/Hpack/ConfigSpec.hs

```diff
@@ -642,6 +645,13 @@ spec = do
         }
         )
 
+    it "allows overriding fields and takes the last value" $ do
+      withPackageConfig_ [i|
+        name: n1
+        name: n2
+        |]
+        (packageName >>> (`shouldBe` "n2"))
```

Comment:
I'm not really sure how to interpret the spec regarding this, but I think that testing for the merge&override case specifically instead of this double definition would be a good idea anyway. :-)